pycontrails/core/flight.py

```diff
@@ -1453,38 +1455,87 @@ def _altitude_interpolation(
     end_na_idxs = np.flatnonzero(end_na)
     na_group_size = end_na_idxs - start_na_idxs
 
-    # Form array of cumulative altitude values if the flight were to climb
-    # at nominal_rocd over each group of nan
+    s = pd.Series(altitude)
 
     if climb_or_descend_at_end:
```

Comment:
You may want to create two helper functions depending on `climb_or_descend_at_end`. So this may look like:

```python
if climb_or_descend_at_end:
   return _altitude_interpolation_climb_descend_end(altitude, s, na_group_size, other_params_needed, ...)
return _altitude_interpolation_climb_descend_middle(altitude, s, na_group_size, other_params_needed, ...)
```